app-backend/tool/src/main/scala/ast/MapAlgebraAST.scala

```diff
@@ -94,15 +107,19 @@ object MapAlgebraAST {
   }
 
   @JsonCodec
-  case class Constant(id: UUID, constant: Double, metadata: Option[NodeMetadata]) extends MapAlgebraLeaf("const") {
+  case class Constant(id: UUID, constant: Double, metadata: Option[NodeMetadata]) extends MapAlgebraLeaf {
+    val `type` = "const"
+
     def sources: Seq[MapAlgebraAST.MapAlgebraLeaf] = List()
 
     def substitute(substitutions: Map[UUID, MapAlgebraAST]): Option[MapAlgebraAST] = Some(this)
   }
 
   /** Map Algebra sources */
   @JsonCodec
-  case class Source(id: UUID, metadata: Option[NodeMetadata]) extends MapAlgebraLeaf("src") {
+  case class Source(id: UUID, metadata: Option[NodeMetadata]) extends MapAlgebraLeaf {
+    val `type` = "src"
```

Comment:
`type` is defined but not used anywhere, is this an orphan or some refactor or will it be required shortly?